app/controllers/FileUploadController.scala

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import config.featureFlags.SfusConfig
+import controllers.actions.AuthAction
+import models.requests.{AuthenticatedRequest, ExportsSessionKeys}
+import play.api.mvc.{Action, AnyContent, Call, MessagesControllerComponents}
+import services.audit.{AuditService, AuditTypes, EventData}
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+
+import javax.inject.Inject
+
+class FileUploadController @Inject()(sfusConfig: SfusConfig, authenticate: AuthAction, mcc: MessagesControllerComponents, auditService: AuditService)
+    extends FrontendController(mcc) {
+
+  def startFileUpload(mrn: String = ""): Action[AnyContent] = authenticate { implicit request =>
+    lazy val sfusLink = s"${sfusConfig.sfusUploadLink}/$mrn"
+
+    def auditData(eori: String, lrn: Option[String], mrn: String, docUploadUrl: String) =
```

Comment:
Any reason for `auditData` to not be placed in `AuditService.audit`? IMO `EventData` should be created by the service and not at Controller-level.

Reply:
Do you mean putting `auditData` in the actual `AuditService` file? Or just in the call?

If the former, don't I need to label my data before it gets sent to the service - because otherwise the service won't know what's what?

Reply:
Ah, OK. I see. `auditService.audit` is the generic method for auditing, used by other controllers as well. Sorry, ignore my comment.